volmdlr/edges.py

```diff
@@ -278,6 +278,203 @@ def split(self, split_point):
+        return self.__class__(self.degree, control_points,
+                              self.knot_multiplicities, self.knots,
+                              self.weights, self.periodic)
+
+    def point_belongs(self, point, abs_tol=1e-10):
+        '''
+        check if a point belongs to the bspline_curve or not
+        '''
+
+        point_dimension = f'Point{self.__class__.__name__[-2::]}'
+
+        def f(x):
+            return (point - getattr(volmdlr, point_dimension)(*self.curve.evaluate_single(x))).norm()
+
+        x = npy.linspace(0, 1, 5)
+        x_init = []
+        for xi in x:
+            x_init.append(xi)
+
+        for x0 in x_init:
+            z = scp.optimize.least_squares(f, x0=x0, bounds=([0, 1]))
+            if z.fun < abs_tol:
+                return True
+        return False
+
+    def merge_with(self, bspline_curve):
+        '''
+        merge successives bspline_curves to define a new one
+        '''
+
+        point_dimension = f'Wire{self.__class__.__name__[-2::]}'
+        wire = getattr(volmdlr.wires, point_dimension)(bspline_curve)
+        ordered_wire = wire.order_wire()
+
+        points, n = [], 10
+        for primitive in ordered_wire.primitives:
+            points.extend(primitive.polygon_points(n))
+        points.pop(n + 1)
+
+        return self.__class__.from_points_interpolation(points, min(self.degree, bspline_curve.degree))
+
+    @classmethod
+    def from_bsplines(cls, bsplines, discretization_points=10):
+        '''
+        define a bspline_curve using a list of bsplines
+        '''
+        point_dimension = f'Wire{cls.__name__[-2::]}'
+        wire = getattr(volmdlr.wires, point_dimension)(bsplines)
+        ordered_wire = wire.order_wire()
+
+        points, degree = [], []
+        for i, primitive in enumerate(ordered_wire.primitives):
+            degree.append(primitive.degree)
+            if i == 0:
+                points.extend(primitive.polygon_points(discretization_points))
+            else:
+                points.extend(primitive.polygon_points(discretization_points)[1::])
+
+        return cls.from_points_interpolation(points, min(degree))
+
+    @classmethod
+    def from_points_approximation(cls, points, degree, **kwargs):
+        '''
+        Bspline Curve approximation through points using least squares method
+        It is better to specify the number of control points
+
+        Parameters
+        ----------
+        points : volmdlr.Point
+            data points
+        degree: int
+            degree of the output parametric curve
+
+        Keyword Arguments:
+            * ``centripetal``: activates centripetal parametrization method. *Default: False*
+            * ``ctrlpts_size``: number of control points. *Default: len(points) - 1*
+
+        Returns
+        -------
+        BSplineCurve
+
+        '''
+
+        curve = fitting.approximate_curve([point.coordinates() for point in points], degree, **kwargs)
```

Comment:
You can remove Point2D/3D.coordinates, because you can use a `*` iterator here.
```suggestion
        curve = fitting.approximate_curve([[*point] for point in points], degree, **kwargs)
```